users/tests/test_user_login.py

```diff
@@ -1,12 +1,32 @@
 from django.test import TestCase
+from django.urls import reverse
 
 from users.models import CustomUser
+from users.models import CustomUserPreferences
 
 
 class TestCustomUserLogin(TestCase):
     def setUp(self):
         self.user_type = CustomUser.UserType.EMPLOYEE.name
-        CustomUser.objects._create_user("testuser@codepoets.it", "testuserpasswd", False, False, self.user_type)
+        self.password = "testuserpasswd"
+        self.user = CustomUser.objects._create_user(
+            email="testuser@codepoets.it",
+            password=self.password,
+            is_staff=False,
+            is_superuser=False,
+            user_type=self.user_type,
+            need_change_password=True,
+        )
+        CustomUserPreferences(user=self.user).save()
+        self.data = {"username": self.user.email, "password": self.password}
+
+    def _set_need_change_password(self):
+        self.user.need_change_password = True
+        self.user.save()
+
+    def _reset_need_change_password(self):
+        self.user.need_change_password = False
+        self.user.save()
```

Comment:
It would be better to have here one method called `_set_need_change_password` and add bool parameter for `need_change_password` property:

```python
    def _set_need_change_password(self, need_change_password: bool):
        self.user.need_change_password = need_change_password
        self.user.save()
```

Also private methods in tests should be **at the end** of test class instead of being at the top. You should always read code as you would read a book. Below I will present you an example of how to properly order methods:

```python
def funA():
    funB()
    funC()
    funD()

def funB():
    funE()

def funE():
    print("funE")

def funC():
    print("funC")

def funD():
    print("funD")
```

Please remember that what I just wrote above is an example for production code. In case of tests order should look like this:



```python
class TestClass(TestCase):
    def setUp(sefl):
        ...

    def test_1(self):
        self._private_method()

    def test_2(self):
        self._private_method_2()

    def _private_method(self):
        ...

    def _private_method_2(self):
        ...
```

Reply:
Ok. Fixed.